[SPARK-23169][INFRA][R] Run lintr on the changes of lint-r script and .lintr configuration

HyukjinKwon opened the pull request:
## What changes were proposed in this pull request?

When running the `run-tests` script, seems we don't run lintr on the changes of `lint-r` script and `.lintr` configuration.

## How was this patch tested?

Jenkins builds

HyukjinKwon committed:
Double check if it really doesn't run the style check

HyukjinKwon committed:
Run lintr on the changes of lint-r script and .lintr configuration

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution/63/
Test PASSed.

HyukjinKwon commented:
retest this please

SparkQA commented:
**[Test build #86430 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86430/testReport)** for PR 20339 at commit [`fe59c8a`](https://github.com/apache/spark/commit/fe59c8aa2b6e66930ef1fade198d35eeb27210ef).

HyukjinKwon committed:
Revert "Double check if it really doesn't run the style check"

This reverts commit fe59c8aa2b6e66930ef1fade198d35eeb27210ef.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution/64/
Test PASSed.

SparkQA commented:
**[Test build #86431 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86431/testReport)** for PR 20339 at commit [`12789c7`](https://github.com/apache/spark/commit/12789c759be18d81335600a1dd54eeb33ce76cf2).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution/65/
Test PASSed.

SparkQA commented:
**[Test build #86432 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86432/testReport)** for PR 20339 at commit [`5ca17ac`](https://github.com/apache/spark/commit/5ca17acfbaa5f2b2682fabe0cfb1d1feaa9a1819).

HyukjinKwon commented:
This was tested:

1. https://github.com/apache/spark/pull/20339/commits/fe59c8aa2b6e66930ef1fade198d35eeb27210ef intentionally added some changes in `.lintr` and `lint-r` to see if triggers the lint:   https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86430/console

2. Added the detection logic to check if it triggers the lintr: https://github.com/apache/spark/pull/20339/commits/12789c759be18d81335600a1dd54eeb33ce76cf2 - https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86431/console



3. Reverted the change of 1. back to leave only 2. change in this PR.

cc @felixcheung and @JoshRosen, could you take a look please?

SparkQA commented:
**[Test build #86430 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86430/testReport)** for PR 20339 at commit [`fe59c8a`](https://github.com/apache/spark/commit/fe59c8aa2b6e66930ef1fade198d35eeb27210ef).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/86430/
Test PASSed.

SparkQA commented:
**[Test build #86431 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86431/testReport)** for PR 20339 at commit [`12789c7`](https://github.com/apache/spark/commit/12789c759be18d81335600a1dd54eeb33ce76cf2).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

SparkQA commented:
**[Test build #86432 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/86432/testReport)** for PR 20339 at commit [`5ca17ac`](https://github.com/apache/spark/commit/5ca17acfbaa5f2b2682fabe0cfb1d1feaa9a1819).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/86431/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/86432/
Test PASSed.
